src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -119,7 +121,10 @@ public void close() throws IOException {
     }
   }
 
-  private static ParquetProperties createParquetProperties() {
+  private static ParquetProperties createParquetProperties(
+      float estimatedUncompressedChunkSizeInBytes) {
+    int pageSizeLimit =
+        (int) MAX_CHUNK_SIZE_IN_BYTES * 2; // (int) (estimatedUncompressedChunkSizeInBytes * 2);
```

Comment:
Why is this commented out? And what's the rationale for the `* 2`?

Reply:
The idea was to estimate size after compression (pageSizeLimit for Parquet)) based on size of data before compression (estimatedUncompressedChunkSizeInBytes). 

> And what's the rationale for the * 2?
It is simply an assumption that the size after compression will be at most twice the size before compression (e.g. for small rows the size after compression will be larger). But I am not sure that this is a valid assumption that's why it is commented out.

Reply:
We may be able to remove this, see https://snowflake.slack.com/archives/C041W9578S1/p1675359412548999?thread_ts=1673361185.095149&cid=C041W9578S1